panoply/datasource.py

```diff
@@ -18,13 +22,80 @@ def log(self, *msgs):
         if "logger" in self.options:
             self.options["logger"](msgs)
         else:
-            print msgs
+            print(msgs)
 
-    def progress(self, loaded, total, msg):
+    def progress(self, loaded, total, msg=''):
         self.fire("progress", {
             "loaded": loaded,
             "total": total,
             "msg": msg
         })
 
 
+def invalidate_token(refresh_url, callback=None,
+                     access_key='access_token', refresh_key='refresh_token'):
+    ''' a decorator used to invalidate the access_token for oauth based
+    data sources.
+    This decorator should be used on every method in the data source that
+    fetches data from the oauth controlled resource, and that relies on a
+    valid access_token in order to operate properly
```

Comment:
please add that we will generate new access token if needed and save it in source[access_key]